client/src/components/browse-recipes/Category.svelte

```diff
@@ -0,0 +1,104 @@
+<script>
+  import { createEventDispatcher, onMount } from 'svelte';
+  import categoriesApi from '@/api/categoriesApi';
+  const dispatch = createEventDispatcher();
+  addEventListener('DOMContentLoaded', event => {
+    let element = document.getElementById('categorySelect');
+
+    element.addEventListener('change', () => {
```

Comment:
big nono, in svelte and other frameworks we don't addEventListeners in this way, all the events you need to register are registered on the element itself, like so

```js
const handleChange = element => {
  dispatch('categoryToggle', {
    value: element.value,
    text: element.options[element.selectedIndex].text,
  });
};
...

<select on:change={handleChange}...>
```